mmdeploy/codebase/mmpose/models/heads/yolox_pose_head.py

```diff
@@ -0,0 +1,157 @@
+
+    flatten_bbox_preds = [
+        bbox_pred.permute(0, 2, 3, 1).reshape(num_imgs, -1, 4)
+        for bbox_pred in bbox_preds
+    ]
+    flatten_bbox_preds = torch.cat(flatten_bbox_preds, dim=1)
+
+    if objectnesses is not None:
+        flatten_objectness = [
+            objectness.permute(0, 2, 3, 1).reshape(num_imgs, -1)
+            for objectness in objectnesses
+        ]
+        flatten_objectness = torch.cat(flatten_objectness, dim=1).sigmoid()
+        cls_scores = cls_scores * (flatten_objectness.unsqueeze(-1))
+
+    scores = cls_scores
+
+    bboxes = bbox_decoder(flatten_priors[None], flatten_bbox_preds,
+                          flatten_stride)
+
+    post_params = get_post_processing_params(deploy_cfg)
+    max_output_boxes_per_class = post_params.max_output_boxes_per_class
+    iou_threshold = cfg.nms.get('iou_threshold', post_params.iou_threshold)
+    score_threshold = cfg.get('score_thr', post_params.score_threshold)
+    pre_top_k = post_params.get('pre_top_k', -1)
+    keep_top_k = post_params.get('keep_top_k', -1)
+
+    priors = torch.cat(self.mlvl_priors)
+    strides = [
+        priors.new_full((featmap_size.numel() * self.num_base_priors, ),
+                        stride)
+        for featmap_size, stride in zip(featmap_sizes, self.featmap_strides)
+    ]
+    strides = torch.cat(strides)
+    kpt_preds = torch.cat([
+        kpt_pred.permute(0, 2, 3, 1).reshape(
+            num_imgs, -1, self.num_keypoints * 2) for kpt_pred in kpt_preds
+    ],
+                          dim=1)
+    flatten_decoded_kpts = self.decode_pose(priors, kpt_preds, strides)
+
+    vis_preds = torch.cat([
+        vis_pred.permute(0, 2, 3, 1).reshape(num_imgs, -1, self.num_keypoints)
+        for vis_pred in vis_preds
+    ],
+                          dim=1).sigmoid()
+
+    result_list = []
+    pred_bbox = bboxes
+    pred_kpts = flatten_decoded_kpts
+    pred_kpts_score = vis_preds
+    pred_score, pred_label = scores.max(2, keepdim=True)
+    nms_result = multiclass_nms(
+        pred_bbox,
+        pred_score,
+        max_output_boxes_per_class,
+        iou_threshold,
+        score_threshold,
+        pre_top_k,
+        keep_top_k,
+        output_index=True)
+
+    keep_indices_nms = nms_result[2]
+    bbox = pred_bbox[:, keep_indices_nms].squeeze(1)
+    label = pred_label[:, keep_indices_nms].squeeze(1)
+    score = pred_score[:, keep_indices_nms].squeeze(1)
+    kpts = pred_kpts[:, keep_indices_nms].squeeze(1)
+    kpts_score = pred_kpts_score[:, keep_indices_nms].squeeze(1)
+
+    # pad_param = batch_img_metas[0].get('img_meta', None)
+    # scale_factor = batch_img_metas[0]['scale_factor']
+    # if pad_param is not None:
+    #     kpts -= kpts.new_tensor([pad_param[2], pad_param[0]])
+    # kpts /= kpts.new_tensor(scale_factor).repeat(
+    #     (1, self.num_keypoints, 1))
+
+    result_list.append([bbox, label, score, kpts, kpts_score])
+
+    return result_list
```

Comment:
since it only has 'background` and `person` two classes, the `label` tensor is useless.